### PR TITLE
fix: sys tests use async/await to allow a fix

### PR DESCRIPTION
**Pull request by DominicKramer**
The system tests have been updated to use async/await.  This is
done to make it easier to see errors in the tests and fix them.

Fixes: #213

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
@DominicKramer it looks like the unit tests are failing - i was going to wait until those were passing to review :)

**Comment by DominicKramer** [comment]
@JustinBeckwith Yes I'm still in the process of resolving some issues that are causing the tests to fail.

Normally, I don't expect a review until I've explicitly added reviewers.  However, I could instead indicate in the title not to review.  What approach do you suggest I should follow that would align with your team's expectations for a PR that doesn't need to be reviewed yet?

**Comment by JustinBeckwith** [comment]
Ahhhh, got it.  Apologies!  Typically I don't open the PR until it's ready for some kind of review 🤷‍♂️  This is my bad though, I should have paid better attention.  I just saw the issue where someone was asking for a release and got eager :)

**Comment by DominicKramer** [comment]
@JustinBeckwith No worries.  It is ready to review now :)

File: system-test/error-reporting.ts [hunk 1]
```diff
@@ -479,82 +487,100 @@ describe('error-reporting', () => {
...
-              errItem && errItem.representative &&
-              errItem.representative.serviceContext &&
-              errItem.representative.serviceContext.service === SERVICE &&
-              errItem.representative.serviceContext.version === VERSION &&
-              messageTest(errItem.representative.message));
-        });
-
-        cb(matchedErrors);
+  async function verifyAllGroups(
+      messageTest: (message: string) => void, maxCount: number,
```
Comment (JustinBeckwith) [inline system-test/error-reporting.ts]: I see `(message: string) => void` used in a few places.  Could we level that up to a type at the top and reuse it?

**Pull request merged**